Doc/c-api/object.rst

```diff
@@ -613,3 +613,19 @@ Object Protocol
 
    .. versionadded:: 3.14
 
+.. c:function:: int PyUnstable_IsImmortal(PyObject *obj)
+
+   This function returns ``1`` if *obj* is :term:`immortal`, and ``0`` otherwise. This function cannot fail.
+
+   Immortal objects don't care about reference counting, thus they no-op calls to :c:func:`Py_INCREF`
+   and :c:func:`Py_DECREF`. Some immutable objects such as literal strings, small integers, or special tuples
+   might be made immortal as an optimization by the interpreter.
+
+   On the :term:`free-threaded <free threading>` build, some objects that support deferred reference counting
+   (see :c:func:`PyUnstable_Object_EnableDeferredRefcount`) might also be immortalized.
```

Comment:
I'm not sure if this information is relevant here? Maybe move it to the glossary?